app/models/metais/document.rb

```diff
@@ -0,0 +1,8 @@
+class Metais::Document < ApplicationRecord
+  self.table_name = 'metais.dokumenty'
+
+  has_paper_trail
+
+  belongs_to :attachable, polymorphic: true
```

Comment:
Tieto polymorficke veci nemam velmi rad, rozbija to referencnu integritu v DB, potom v dokumentacii nevieme pekne linkovat medzi tabulkami poriadne. Mimo Rails sveta to ziadny tool nepozna. Je nejaky specialny dovod preco takto ci len sme si to chceli zjednodusit?

Reply:
Radsej spravme 2 modely na dokumenty (pre isvs a projekty) a k nim separe tabulky, aspon sa nam to nebude miesat. Mozu dedit vacsinu spravania.

Reply:
Pri tom paper_trail sme si to cheli zjednodušiť, že nebude treba kódiť žiadnu logiku okolo toho. Je pravda, že trošku to asi bude špatiť tej datahub DB, keďže uložený `item_type` je názov modelu z Ruby, čo je väčšinou anglický preklad tabuľky v datahube. Nedá sa to tam nastaviť bez polymorfizmu, že by sme pre každý sledovaný model mali jednu tabuľku verzií.

Ak je to problém, vieme to robiť tak, že zo stiahnutých dát vyrobíme vždy novú inštanciu a tú porovnáme s tou, ktorú už možno máme v DB. Ak v nich bude rozdiel, vieme podobným spôsobom ako ten paper_trail uložiť starú verziu do nejakej tabuľky `<model>_versions`. Prípadne vieme nejakým diffom ukladať iba to, čo sa tam zmenilo medzi verziami. Oproti paper_trail prídeme o funkcionalitu, že vieme jednoducho pozerať objekt v jeho starších verziách. Ale to si myslím, že ani nemáme záujem používať.

Reply:
@celuchmarek Aha, paper trail ma len polymorficky model? Tak to si nekazme a vyrobme si vlastne a porovname jsony a ked bude iny zapisujeme novy, pricom v modeli hore si drzime aktualny pripadne staci link na najnovsiu verziu.

Reply:
Spravil som niečo na spôsob RF, kde máme tiež rôzne verzie hodnotení. Tuto sú hlavné tabuľky s uuid a odkazom na `latest_version`.